git_theta/checkpoints.py

```diff
@@ -157,3 +157,23 @@ def get_checkpoint(checkpoint_type: str) -> Checkpoint:
     """
     discovered_plugins = entry_points(group="git_theta.plugins.checkpoints")
     return discovered_plugins[checkpoint_type].load()
+
+
+def get_checkpoint_class() -> Checkpoint:
```

Comment:
The main point of this is sharing how the checkpoint type is inferred between clean and smudge filters right?

Would it be better to share a `get_checkpoint_type` function instead of sharing via a wrapper around the actual checkpoint getter? You get code reuse without having two ways to get checkpoints moving forward.

Something like:

```python
def get_checkpoint_type(checkpoint_type: Optional[str] = None) -> str:
  return checkpoint_type or os.environ.get("GIT_THETA_CHECKPOINT_TYPE") or "pytorch"
```

Then the clean filter, the smudge filter, and the add binary can all do things the same way i.e.

```python
  # Clean and Smudge would call this with no params.
  checkpoint_type = checkpoints.get_checkpoint_type(args.checkpoint_type)
  checkpoint = checkpoints.get_checkpoint(checkpoint_type)
```

Could also be something like `get_checkpoint_name` instead of type if we want.

Reply:
Any reason to leave `get_checkpoint_type: Optional[str] --> str` and `get_checkpoint: str --> Checkpoint` as two separate functions? New revision has them combined into `get_checkpoint_handler: Optional[str] --> Checkpoint` that gets called in add, clean, and smudge.

Reply:
You're right there isn't much of a point to having 2 user facing functions, but wrt to testing we should probably have a private function that gets the checkpoint type instead of in-lining the defaulting logic.

There should be at least 3 tests:
* `test_get_checkpoint_prefers_user_input` where the argument and env var are set and we assert the input is returned
* `test_get_checkpoint_uses_env_variable` where the argument is none and the env is set. We assert the return is the value in the env variable
* `test_get_checkpoint_default` where the argument is None, (maybe 2 versions one where env is `""` and one where it is unset), and assert we get the default.

Basically these tests assert we are overriding config options in the right order and will be useful if things ever change/as an outline for how all configuration cascades should work.

Reply:
I wrote similar tests for the update plugin handlers here https://github.com/r-three/git-theta/blob/fa088d3b897e4edfa0f64fb5e3431d3d3658f702/tests/updates/base_test.py

IIRC you mentioned you hadn't used pytest much, so i figured I'd share how to set env vars for a single test via fixtures

Reply:
Nice, thanks for the example. Tests for precedence of user input, environment variable, and default are added.